src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -1983,6 +2010,21 @@ private async Task StopAsyncCore()
             _stopTcs!.TrySetResult(null);
         }
 
+        // Marks ConnectionState as no longer in use. State checked by StopAsync so that we don't access properties that might be disposed already
+        public void Complete()
+        {
+            lock (_lock)
+            {
+                if (_stopTcs != null || _complete)
+                {
+                    return;
+                }
+
+                _messageBuffer?.Dispose();
```

Comment:
Having two places we dispose things feels prone to failure. Can we have `StopAsyncCore` call into this assuming we still need to dispose the `_messageBuffer` when the transport stops on its own?